core/includes/install.core.inc

```diff
@@ -1794,6 +1794,13 @@ function install_check_requirements($install_state) {
 function _install_configure_form($form, &$form_state, &$install_state) {
   include_once BACKDROP_ROOT . '/core/includes/locale.inc';
 
+  if (!empty(ini_get('sendmail_from'))) {
+    $site_email = ini_get('sendmail_from');
+  }
+  else {
+    $site_email = "no-reply@" . $_SERVER['SERVER_NAME'];
```

Comment:
Server name is not going to be valid for email addresses, as it can contain `www` and other subdomains. I have a PR with a function to generate valid email addresses, we can use that in which ever approach we take (no need to spend time on it) :)